107/src/main/java/org/ehcache/jsr107/Eh107Cache.java

```diff
@@ -525,7 +525,28 @@ public void deregisterCacheEntryListener(CacheEntryListenerConfiguration<K, V> c
 
   @Override
   public java.util.Iterator<Cache.Entry<K, V>> iterator() {
-    return new Iterator<K, V>(ehCache);
+    checkClosed();
+
+    final java.util.Iterator<org.ehcache.Cache.Entry<K, V>> specIterator = jsr107Cache.specIterator();
+    return new java.util.Iterator<Entry<K, V>>() {
+      @Override
+      public boolean hasNext() {
+        checkClosed();
+        return specIterator.hasNext();
+      }
+
+      @Override
+      public Entry<K, V> next() {
+        checkClosed();
+        return new WrappedEhcacheEntry<K, V>(specIterator.next());
```

Comment:
Yes, good spot. I'll fix that and make sure there is test coverage for this particular case.